csrc/fusion_profiler.cpp

```diff
@@ -628,15 +628,11 @@ void FusionProfiler::stop() {
     NVFUSER_CUPTI_SAFE_CALL(cuptiActivityDisable(CUPTI_ACTIVITY_KIND_DRIVER));
     NVFUSER_CUPTI_SAFE_CALL(
         cuptiActivityDisable(CUPTI_ACTIVITY_KIND_EXTERNAL_CORRELATION));
+    // This will be populated by the following `cuptiActivityFlushAll` call.
     fp->kernel_profiles_.reserve(fp->segments_.size());
-    fprof.kernel_profiles.resize(fp->segments_.size());
-
     NVFUSER_CUPTI_SAFE_CALL(cuptiActivityFlushAll(0));
 
-    NVF_CHECK(
-        fp->kernel_profiles_.size() >= fp->segments_.size(),
```

Comment:
This assumption no longer holds with this PR. We may run fewer kernels than segments.